nautobot/circuits/views.py

```diff
@@ -59,23 +60,38 @@ def get_extra_context(self, request, instance):
 
 class CircuitTerminationUIViewSet(
     view_mixins.ObjectDetailViewMixin,
+    view_mixins.ObjectListViewMixin,
     view_mixins.ObjectEditViewMixin,
     view_mixins.ObjectDestroyViewMixin,
+    view_mixins.ObjectBulkDestroyViewMixin,
+    view_mixins.ObjectBulkCreateViewMixin,
     view_mixins.ObjectChangeLogViewMixin,
     view_mixins.ObjectNotesViewMixin,
 ):
+    action_buttons = ("import", "export")
+    filterset_class = filters.CircuitTerminationFilterSet
     form_class = forms.CircuitTerminationForm
     queryset = CircuitTermination.objects.all()
     serializer_class = serializers.CircuitTerminationSerializer
+    table_class = tables.CircuitTerminationTable
 
     def get_object(self):
         obj = super().get_object()
         if self.action in ["create", "update"] and "circuit" in self.kwargs:
             obj.circuit = get_object_or_404(Circuit, pk=self.kwargs["circuit"])
         return obj
 
-    def get_return_url(self, request, obj):
-        return obj.circuit.get_absolute_url()
+    def get_return_url(self, request, obj=None):
+        # First, see if `return_url` was specified as a query parameter or form data. Use this URL only if it's
+        # considered safe.
+        query_param = request.GET.get("return_url") or request.POST.get("return_url")
+        if query_param and is_safe_url(url=query_param, allowed_hosts=request.get_host()):
+            return query_param
+
+        if obj is not None and obj.present_in_database and obj.pk:
```

Comment:
Are `obj is not None` and `obj.pk` and `obj.present_in_database` all necessary for this check?

Reply:
It's copied logic from the base viewset mixin. IIRC it's there to prevent us from accidentally redirecting to an object that's in the process of being deleted.